include/cuda/std/detail/libcxx/include/__mdspan/layout_right.h

```diff
@@ -170,18 +170,14 @@ class layout_right::mapping {
         * TODO: check precondition
         * __other.required_span_size() is a representable value of type index_type
         */
-       #ifndef __CUDA_ARCH__
-       size_t __stride = 1;
-       for(rank_type __r=__extents.rank(); __r>0; __r--) {
-         #ifndef _LIBCUDACXX_NO_EXCEPTIONS
```

Comment:
I think we should keep the exceptions path in the code base, so that we don't miss this spot as one that should throw if/when we decide to support a mode where we throw exceptions.

Reply:
So the real reason I have removed this is because NV_IF_TARGET does not work with `#ifdef` inside. So we would need to replicate the whole code block or extract it into a function.

Therefore I would rather leave that as is and fix that issue when it arrives

Reply:
I have added a separate macro, that dispatches to an assert without exceptions and an runtime error otherwise